app/src/gui/components/notebook/record_table.tsx

```diff
@@ -358,16 +345,93 @@ function RecordsTable(props: RecordsTableProps) {
         },
       ];
 
+  // Fetch the current user when the component mounts
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        const userId = await getCurrentUserId(project_id);
+        setCurrentUser(userId);
+      } catch (error) {
+        console.error('Error fetching user ID:', error);
+      }
+    };
+    fetchUser();
+  }, [project_id]);
+
+  useEffect(() => {
+    if (!rows || rows.length === 0 || !currentUser) {
+      return;
+    }
```

Comment:
If the rows length is zero, shouldn't it call the onRecordsCountChange(0)?

Reply:
DOne